components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -94,22 +95,38 @@ public void addPolicy(String policyLevel, Policy policy) throws APIMgtDAOExcepti
             connection = DAOUtil.getConnection();
 
             //TODO : instead of checking policyLevel, check class type, and remove passing policy level to here
+            if (policy instanceof APIPolicy) {
+                String type = policy.getDefaultQuotaPolicy().getType();
+                Limit limit = policy.getDefaultQuotaPolicy().getLimit();
+                if (type.equals(PolicyConstants.REQUEST_COUNT_TYPE)) {
+                    if (limit instanceof RequestCountLimit) {
+                        addAPIPolicy(connection, policy.getPolicyName(), policy.getDisplayName(),
+                                policy.getDescription(), policy.getDefaultQuotaPolicy().getType(), 10,
+                                policy.getDefaultQuotaPolicy().getLimit().getUnitTime(),
```

Comment:
I think the value 10 is here for bandwidth based throttling right? Can we add a todo there if bandwidth based one is not implemented